sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,65 +31,248 @@
 
 import operator
 import pandas
+import numpy
+import sdc
 
 from numba import types
-from numba.extending import (overload, overload_method, overload_attribute)
 from numba.errors import TypingError
-
-from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
+from sdc.hiframes.pd_dataframe_ext import DataFrameType
 from sdc.utils import sdc_overload_method
+from sdc.datatypes.common_functions import TypeChecker
+from sdc.config import config_pipeline_hpat_default
+
+
+if not config_pipeline_hpat_default:
+    def sdc_pandas_dataframe_drop_codegen(func_name, func_args, df, drop_cols):
+        """
+        Input:
+        df.drop(columns='M', errors='ignore')
+
+        Func generated:
+        def sdc_pandas_dataframe_drop_impl(df, labels=None, axis=0, index=None, columns=None, level=None, inplace=False,
+         errors="raise"):
+            if errors == "raise":
+              raise ValueError("The label M is not found in the selected axis")
+            new_col_A_data_df = get_dataframe_data(df, 0)
+            new_col_B_data_df = get_dataframe_data(df, 1)
+            new_col_C_data_df = get_dataframe_data(df, 2)
+            return pandas.DataFrame({"A": new_col_A_data_df, "B": new_col_B_data_df, "C": new_col_C_data_df})
+
+        """
+        indent = 4 * ' '
+        df_columns_indx = {col_name: i for i, col_name in enumerate(df.columns)}
+        saved_df_columns = [column for column in df.columns if column not in drop_cols]
+        func_definition = [f'def sdc_pandas_dataframe_{func_name}_impl({", ".join(func_args)}):']
+        func_text = []
+        column_list = []
+
+        for label in drop_cols:
+            if label not in df.columns:
+                func_text.append(f'if errors == "raise":')
+                func_text.append(f'  raise ValueError("The label {label} is not found in the selected axis")')
+                break
+
+        for column in saved_df_columns:
+            func_text.append(f'new_col_{column}_data_{"df"} = get_dataframe_data({"df"}, {df_columns_indx[column]})')
+            column_list.append((f'new_col_{column}_data_df', column))
+
+        data = ', '.join(f'"{column_name}": {column}' for column, column_name in column_list)
+        # TODO: Handle index
+        func_text.append(f"return pandas.DataFrame({{{data}}})\n")
```

Comment:
```suggestion
        func_text.append(f"return pandas.DataFrame(data={{{data}}}, index=df._index)\n")
```

I think it should work fine

Reply:
Unboxing for DataFrame index is not implemented yet, so it would not work properly